modules/standard/IO.chpl

```diff
@@ -51,47 +51,63 @@ For example, the following program opens a file and writes an integer to it:
 
 .. code-block:: chapel
 
-  // open the file "test-file.txt" for writing, creating it if
-  // it does not exist yet.
-  var myFile = try! open("test-file.txt", ioMode.cw);
+  try {
+    // open the file "test-file.txt" for writing, creating it if
+    // it does not exist yet.
+    var myFile = open("test-file.txt", ioMode.cw);
+
+    // create a fileWriter starting at file offset 0
+    // (start and end offsets can be specified when creating the
+    // fileWriter)
+    var myFileWriter = myFile.writer();
 
-  // create a fileWriter starting at file offset 0
-  // (start and end offsets can be specified when creating the
-  // fileWriter)
-  var myFileWriter = try! myFile.writer();
+    var x: int = 17;
 
-  var x: int = 17;
+    // This function will write the human-readable text version of x;
+    // binary I/O is also possible.
+    myFileWriter.write(x);
 
-  // This function will write the human-readable text version of x;
-  // binary I/O is also possible.
-  try! myFileWriter.write(x);
+    // Now test-file.txt contains:
+    // 17
 
-  // Now test-file.txt contains:
-  // 17
+  } catch e: Error {
+    // These functions will throw errors under specific circumstances.  The
+    // possible errors vary depending on arguments used and the current state of
+    // the file system.  See the documentation for the individual functions and
+    // the spec chapter on Error Handling for more information.
```

Comment:
Should it say something about how code in production should generally handle the error?

Here is an alternative attempt at this comment that you can use if you like it better.

> Generally speaking, the I/O functions throw errors. Handling these errors is application-dependent and is left out of this example for brevity. Please see the documentation for individual functions for more details about errors that they can throw.